src/runmode-dpdk.c

```diff
@@ -0,0 +1,1274 @@
+{
+    *if_root = ConfSetRootIfaceNode(dpdk_node_name, iface);
+    *if_default = ConfSetDefaultNode(dpdk_node_name);
+
+    if (*if_root == NULL && *if_default == NULL) {
+        SCLogError(SC_ERR_DPDK_CONF,
+                "unable to find DPDK config for "
+                "interface \"%s\" or \"default\", using default values",
+                iface);
+        SCReturnInt(-ENODEV);
+    }
+
+    /* If there is no setting for current interface use default one as main iface */
+    if (*if_root == NULL) {
+        *if_root = *if_default;
+        *if_default = NULL;
+    }
+
+    SCReturnInt(0);
+}
+
+int ConfigSetIface(DPDKIfaceConfig *iconf, const char *entry_str)
+{
+    int retval;
+
+    if (entry_str == NULL || entry_str[0] == '\0') {
+        SCLogError(SC_ERR_INVALID_VALUE, "Passed value is NULL");
+        errno = EINVAL;
+        SCReturnInt(-errno);
+    }
+
+    retval = rte_eth_dev_get_port_by_name(entry_str, &iconf->port_id);
+    if (retval < 0) {
+        SCLogError(SC_ERR_DPDK_CONF, "Name of the interface (%s) not valid", entry_str);
+        errno = -retval;
+        SCReturnInt(retval);
+    }
+
+    strlcpy(iconf->iface, entry_str, sizeof(iconf->iface));
+    SCReturnInt(0);
+}
+
+int ConfigSetThreads(DPDKIfaceConfig *iconf, const char *entry_str)
+{
+    const char *active_runmode = RunmodeGetActive();
+
+    if (active_runmode && !strcmp("single", active_runmode)) {
+        iconf->threads = 1;
+        SCReturnInt(0);
+    }
+
+    if (entry_str == NULL) {
+        SCLogError(SC_ERR_INVALID_VALUE, "Passed value is NULL");
+        errno = EINVAL;
+        SCReturnInt(-errno);
+    }
+
+    if (strcmp(entry_str, "auto") == 0) {
+        iconf->threads = (int)rte_lcore_count() - 1;
+        SCReturnInt(0);
+    }
+
+    if (StringParseInt32(&iconf->threads, 10, 0, entry_str) < 0) {
+        SCLogError(SC_ERR_INVALID_VALUE, "Value %s can not be parsed to int", entry_str);
+        errno = EINVAL;
+        SCReturnInt(-errno);
+    }
+
+    if (iconf->threads < 1 || iconf->threads > ((int)rte_lcore_count() - 1)) {
+        SCLogError(SC_ERR_INVALID_VALUE, "Invalid number of threads");
+        errno = ERANGE;
+        SCReturnInt(-errno);
+    }
+
+    SCReturnInt(0);
+}
+
+int ConfigSetRxQueues(DPDKIfaceConfig *iconf, const char *entry_str)
+{
+    if (entry_str == NULL) {
+        SCLogError(SC_ERR_INVALID_VALUE, "Passed value is NULL");
+        errno = EINVAL;
+        SCReturnInt(-errno);
+    }
+
+    if (strcmp(entry_str, "auto") == 0) {
+        iconf->nb_rx_queues = iconf->threads;
+        SCReturnInt(0);
+    }
+
+    if (StringParseUint16(&iconf->nb_rx_queues, 10, 0, entry_str) < 0) {
+        SCLogError(SC_ERR_INVALID_VALUE, "Value %s can not be parsed to int", entry_str);
+        errno = EINVAL;
+        SCReturnInt(-errno);
+    }
+
+    if (iconf->nb_rx_queues < 1 || iconf->nb_rx_queues > RTE_MAX_QUEUES_PER_PORT) {
```

Comment:
RTE_MAX_QUEUES_PER_PORT is suggested MAX and not the real max for the PMD in use. Please re-work to correctly identify the MAX queues for a NIC and compare.

Reply:
use proper pararenthesis.